peps/pep-0735.rst

```diff
@@ -633,7 +633,311 @@ Footnotes
+behavior more precisely.
+Furthermore, these values can be declared in ``.npmrc`` files, allowing
+per-user and per-project configurations to control installation behaviors.
+
+Ruby & Ruby Gems
+----------------
+
+Ruby projects may or may not be intended to produce packages ("gems") in the
+Ruby ecosystem. In fact, the expectation is that most users of the langauge do
+not want to produce gems and have no interest in producing their own packages.
+Many tutorials do not touch on how to produce packages, and the toolchain never
+requires user code to be packaged for supported use-cases.
+
+Ruby splits requirement specification into two separate files.
+
+- ``Gemfile``: a dedicated file which only supports requirement data in the form
+  of dependency groups
+- ``<package>.gemspec``: a dedicated file for declaring package (gem) metadata
+
+The ``bundler`` tool, providing the ``bundle`` command, is the primary interface
+for using ``Gemfile`` data.
+
+The ``gem`` tool is responsible for building gems from ``.gemspec`` data, via the
+``gem build`` command.
+
+Gemfiles & bundle
+'''''''''''''''''
+
+A `Gemfile <https://bundler.io/v1.12/man/gemfile.5.html>`__ is a ruby file
+containing ``gem`` directives enclosed in any number of ``group`` declarations.
+``gem`` directives may also be used outside of the ``group`` declaration, in which
+case they form an implicitly unnamed group of dependencies.
+
+For example, the following ``Gemfile`` lists ``rails`` as a project dependency.
+All other dependencies are listed under groups:
+
+.. code-block:: ruby
+
+    source 'https://rubygems.org'
+
+    gem 'rails'
+
+    group :test do
+      gem 'rspec'
+    end
+
+    group :lint do
+      gem 'rubocop'
+    end
+
+    group :docs do
+      gem 'kramdown'
+      gem 'nokogiri'
+    end
+
+If a user executes ``bundle install`` with these data, all groups are installed.
+Users can deselect groups by creating or modifying a bundler config in ``.bundle/config``, either manually or via the CLI.
+For example, ``bundle config set --local without 'lint:docs'``.
+
+It is not possible, with the above data, to exclude the top-level use of the ``'rails'`` gem or to refer to that implicit grouping by name.
+
+gemspec and packaged dependency data
+''''''''''''''''''''''''''''''''''''
+
+A `gemspec file <https://guides.rubygems.org/specification-reference/>`__ is a
+ruby file containing a `Gem::Specification
+<https://ruby-doc.org/stdlib-3.0.1/libdoc/rubygems/rdoc/Gem/Specification.html>`__ instance declaration.
+
+Only two fields in a ``Gem::Specification`` pertain to package dependency data.
+These are ``add_development_dependency`` and ``add_runtime_dependency``.
+A ``Gem::Specification`` object also provides methods for adding dependencies
+dynamically, including ``add_dependency`` (which adds a runtime dependency).
+
+Here is a variant of the current ``rails.gemspec`` file at time of writing, with many fields removed or shortened to simplify:
```

Comment:
Unless specified otherwise, everything is "current" and "at time of writing", so we can skip them (unless, for example, something is just about to change).

```suggestion
Here is a variant of the ``rails.gemspec`` file, with many fields removed or shortened to simplify:
```
Please also wrap this and other lines to ~80 chars where possible.